tests/core/openai/_utils/test_handle_stream.py

```diff
@@ -0,0 +1,183 @@
+    tool_call = ChoiceDeltaToolCall(
+        index=0,
+        id=None,
+        function=ChoiceDeltaToolCallFunction(
+            arguments='{"title": "The Name of the Wind", "author": "Patrick Rothfuss"}',
+            name=None,
+        ),
+        type="function",
+    )
+    return [
+        ChatCompletionChunk(
+            id="id",
+            choices=[
+                Choice(delta=ChoiceDelta(content="content", tool_calls=None), index=0)
+            ],
+            created=0,
+            model="gpt-4o",
+            object="chat.completion.chunk",
+        ),
+        ChatCompletionChunk(
+            id="id",
+            choices=[
+                Choice(
+                    delta=ChoiceDelta(
+                        content=None,
+                        tool_calls=[new_tool_call],
+                    ),
+                    index=0,
+                )
+            ],
+            created=0,
+            model="gpt-4o",
+            object="chat.completion.chunk",
+        ),
+        ChatCompletionChunk(
+            id="id",
+            choices=[
+                Choice(
+                    delta=ChoiceDelta(
+                        content=None,
+                        tool_calls=[tool_call],
+                    ),
+                    index=0,
+                )
+            ],
+            created=0,
+            model="gpt-4o",
+            object="chat.completion.chunk",
+        ),
+        ChatCompletionChunk(
+            id="id",
+            choices=[
+                Choice(
+                    delta=ChoiceDelta(
+                        content=None,
+                        tool_calls=[new_tool_call],
+                    ),
+                    index=0,
+                )
+            ],
+            created=0,
+            model="gpt-4o",
+            object="chat.completion.chunk",
+        ),
+        ChatCompletionChunk(
+            id="id",
+            choices=[
+                Choice(
+                    delta=ChoiceDelta(
+                        content=None,
+                        tool_calls=[tool_call],
+                    ),
+                    index=0,
+                )
+            ],
+            created=0,
+            model="gpt-4o",
+            object="chat.completion.chunk",
+        ),
+        ChatCompletionChunk(
+            id="id",
+            choices=[
+                Choice(
+                    delta=ChoiceDelta(content=None, tool_calls=None),
+                    finish_reason="tool_calls",
+                    index=0,
+                )
+            ],
+            created=0,
+            model="gpt-4o",
+            object="chat.completion.chunk",
+        ),
+    ]
+
+
+def test_handle_stream(mock_chunks: list[ChatCompletionChunk]) -> None:
+    """Tests the `handle_stream` function."""
+
+    result = [
+        t for t in handle_stream((c for c in mock_chunks), tool_types=[FormatBook])
+    ]
+    assert len(result) == 3
+    assert result[0][1] is None
```

Comment:
for these types of assertions, add comment so its more clear what is being tested

Reply:
Done.